src/server/lobby-server.ts

```diff
@@ -1,65 +1,72 @@
-import * as cluster from 'cluster';
+import * as express from 'express';
 import { Server }  from 'http';
+import { Worker } from 'cluster';
 import { GameServer } from './game-server';
-import * as SocketIO from 'socket.io';
 import { Game } from '../core/game';
-import { GAME_LOBBY_COUNTDOWN, NEW_PLAYER_JOINED } from './server-interface'
+import { GAME_LOBBY_COUNTDOWN, GAME_CODE_LENGTH } from './server-interface'
 
 export class LobbyServer {
-  players: SocketIO.Socket[];
-  io: SocketIO.Server;
   server: Server;
-  gameStartTimer: number; // Timeout id
+  app: express.Express;
+  gameStartTimer: number = null; // Timeout id
+  playersInRandLobby: number = 0; // Number of players sitting in random lobby
+  currentRandLobby: string; // Unique hash for lobby
+  gameServer: GameServer;
 
-
-  constructor(server: Server) {
+  constructor(server: Server, app: express.Express, gameServer: GameServer) {
+    this.app = app;
     this.server = server;
-    this.io = SocketIO(this.server);
-    this.refreshLobby();
-    this.io.on('connection', this.onConnection.bind(this));
+    this.gameServer = gameServer;
+    this.refreshRandGame();
   }
-  
-  onConnection(socket: SocketIO.Socket) {
-    this.players.push(socket);
-    socket.on('disconnect', () => this.onDisconnection(this.players.length-1));
-    console.log("Players in lobby: ", this.players.length);
 
-    this.resetTimer();
-    for (var playerSocket of this.players) {
-      playerSocket.emit(NEW_PLAYER_JOINED, this.players.length);
-    }
-    if (this.players.length > Game.settings.maxPlayers) {
-      // Assuming we won't go from max-1 players to max+1 players
-      console.log("CRITICAL ERROR: too many players");
-    } else if (this.players.length == Game.settings.maxPlayers) {
-      this.startGame();
-    } else if (this.players.length >= Game.settings.minPlayers) {
-      this.gameStartTimer = setTimeout(this.startGame.bind(this), GAME_LOBBY_COUNTDOWN);
+  // Returns new lobby's hashcode
+  createNewLobby() : string {
+    const alphabet = "abcdefghijklmnopqrstuvwxyz0123456789"
```

Comment:
Don't see how it would be more LOC, less clear, or less time.  Its less code, less to maintain, already tested, and clearer (it also shows that we want short ids), but  ¯\_(ツ)_/¯ up to you
```js
import * as shortid from 'shortid';
...
createNewGameHash() : string { return shortid.generate(); }
```